packages/api/src/car.js

```diff
@@ -109,16 +114,22 @@ export async function carPost (request, env, ctx) {
     local: blob.size > LOCAL_ADD_THRESHOLD
   })
 
+  /** @type {ReturnType<toPins>} */
+  let pins
   // Retrieve current pin status and info about the nodes pinning the content.
-  const { peerMap } = await env.cluster.status(cid)
+  // Keep querying Cluster until one of the nodes reports something other than
+  // Unpinned i.e. PinQueued or Pinning or Pinned.
+  while (true) {
```

Comment:
Can we become prone to Timeouts from CF with this?

I think we can probably add a `p-timeout` with a reasonable timeout for pooling. This will mean we might still present the state we are trying to fix to the user, but at least we will not throw a timeout error when the file is already stored properly